### DISCUSSION TITLE
Custom Icons

### DISCUSSION THREAD

**Discussion by aidanweltner** [livewire/flux#294] [Feature Requests] [answered]
It would be awesome to have an API for custom icons. Something where you could configure a namespace for a custom library and load them in. I.e:

`<flux:button icon="custom:solid.bandcamp">Follow Us on Bandcamp</flux:button>`

I was able to hack custom icons together just by copying the attributes and classes but it would still be really cool to be able to have a slick API to make this happen. I often find that Heroicons is missing icons when not working on tech products. Thanks for the consideration!

**Comment by donnyxray** [comment]
I was hoping for [blade icons](https://blade-ui-kit.com/blade-icons) support.

**Reply by AlexanderFalkenberg** [reply]
Also #232

**Comment by mauritskorse** [comment]
I have created a laravel package for adding custom icons to flux: https://github.com/Ympact/flux-icons
It's the first version, so there might still be some bugs ;)
It currently only has the [Tabler](https://tabler.io/icons) icon-set, but other vendors can easily be added, although some tweaking might be necessary to make the icons properly work in Flux.

**Reply by joshhanley** [reply]
@mauritskorse FYI you may be in violation of the Flux licence. Specifically this clause in the terms of service: Creating a public repository of Flux components or derivatives.

Can I suggest making your repo private for now, and I will seek clarification from Caleb as to what licence the public livewire/flux repo is under? As currently there is nothing published in the repo, so the assumption would be that it falls under Fluxs licence and terms of use.

**Reply by mauritskorse** [reply]
Ah, sure. Will do.
I hadn't checked those, since since the flux icons component is in the public repo of Flux.

**Reply by joshhanley** [reply]
@mauritskorse All good! I've messaged Caleb so will see what he says.

**Reply by joshhanley** [reply]
@mauritskorse Caleb said the licence isn't MIT yet. But he said all good to keep working on and release your icons package 🙂

**Reply by mauritskorse** [reply]
Thanks :)

**Comment by RyanPaiva56** [comment]
IMO, it would be cool to see Caleb be able to focus on the UI and not delve into maintaining a list of supported icon sources. Heroicons is enough for a base. That being said, I think having an ecosystem section for recommended packages would be very cool, and put @mauritskorse's package on that list. I love the idea of having more custom icons and it being a different package makes sense.

**Comment by calebporzio** [comment, marked as answer]
Going to close this as we support lucide now and have docs for custom svgs: https://fluxui.dev/components/icon#lucide-icons
